### ISSUE TITLE
Support for custom $(Configuration)|$(Platform) conditions in VS property-sheets

### ISSUE THREAD

**Issue by g-glowacki** [Caphyon/clang-power-tools#81]
![screenshot](https://user-images.githubusercontent.com/28043060/32325817-d50f0f88-bfd0-11e7-85d0-537d03730c18.png)

I don't know if it's a problem with my configuration, but i can't configure includes...
I have two VS 2015 opened (different git branches for the same code), and on one of them the include appears, on the second one doesn't. Furthermore, on the first one I can use power-tools, but on the second (without includes) execution of compile/tidy gives error:
```
Start Clang Tidy

[ INFO ] WILL PROCESS PROJECTS: 
	xxx.vcxproj
[ INFO ] 1. PROCESSING PROJECT D:\xxx.vcxproj
Did not expect that
At C:\Users\gglowacki\AppData\Local\Microsoft\VisualStudio\14.0\Extensions\h3be4nct.gu5\clang-build.ps1:802 char:9
+         throw "Did not expect that"
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Did not expect that:String) [], RuntimeException
    + FullyQualifiedErrorId : Did not expect that
 

Done Clang Tidy


```
Clang-power-tools 1.5.0

**Comment by cpp-red-lion** [comment]
Hi,

Project `Include Directories` should be automatically extracted from your `*.vcxproj` or VS property-sheets.
There is no need to configure them manually.
The _global setting_ you are referring to is there for historical reasons and has been *removed* in version *1.4*, to avoid further confusion.

**Comment by cpp-red-lion** [comment]
If you have any issues with a specific project of yours (related to `Include Directories`) please include a sample `*.vcxproj` reproducing the problem.

Thanks.

**Comment by g-glowacki** [comment]
unfortunately the new version doesn't work for me... It gives error as mentioned in the beginning.

**Comment by cpp-red-lion** [comment]
I'm sorry about that.
Can you attach your project config (or a similar/simplified project) here, so we can debug your issue ?
That would really help. We are not able to reproduce the issue with any of our test projects...

**Comment by g-glowacki** [comment]
unfortunately i can't, but our project is a big solution (about 50) projects with many references inside. But as i see the exception thrown inside the ps script means, that there can't be more then one 
```
      if ($inheritedNodes.Count -gt 1)
```
I think You should consider looking here.

**Comment by cpp-red-lion** [comment]
OK, good hint. 
We'll try that, but it's much more difficult to investigate blindly without a proper test case.
I'll dig into it and look for a way to patch it.

I'm counting on you to test the script afterwards to see if the problem was fixed.

**Comment by g-glowacki** [comment]
with pleasure:)

**Comment by hero101111** [comment]
HI @g-glowacki ,

There was a limitation in the PS scripts: it didn't allow for conditioned elements in property sheets.
It assumed the config/platform conditioning was handled in the .vcxproj xml.

My guess is that you have such conditioned elements in your property sheets.
[I've just fixed this limitation. ](https://github.com/Caphyon/clang-power-tools/commit/78a96b3339562ed2f511997c56efe0c594aebbf3)

Please test the latest `clang-build.ps1` script from GitHub and let us know if it fixes your issue.

--------------------------

Note that this fix is currently live only for the github repository.
The latest Visual Studio extension (1.5.0) does not contain the fix. It will be included in the next release.

Until then, as a workaround, you can manually replace clang-build.ps1 in the VS extension directory with the latest from github.

To locate the extension directory search for clang-build.ps1 in %LOCALAPPDATA%

**Comment by g-glowacki** [comment]
I manually replaced the PowerShell script, but it still doesn't work. 
```
Start Clang Tidy

[ INFO ] WILL PROCESS PROJECTS: 
	SundialDICOMNetReports.vcxproj
[ INFO ] 1. PROCESSING PROJECT D:\workspace\econib\main\SundialLibraries\SundialDICOMNetReports\SundialDICOMNetReports.vcxproj
Did not expect that
At C:\Users\gglowacki\AppData\Local\Microsoft\VisualStudio\14.0\Extensions\24rhmtuq.ud0\clang-build.ps1:802 char:9
+         throw "Did not expect that"
+         ~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : OperationStopped: (Did not expect that:String) [], RuntimeException
    + FullyQualifiedErrorId : Did not expect that
 

Done Clang Tidy

```

**Comment by hero101111** [comment]
Did you restart Visual Studio after replacing the PS script?

**Comment by g-glowacki** [comment]
Yes, i restarted it.

**Comment by hero101111** [comment]
I'm afraid we're at a loss here. Can't spot any other potential cause for this issue.
We need a testable project. If you're unable to upload them here, you can send the project/prop sheets to _gabriel **dot** diaconita **at** caphyon **dot** com_

**Comment by hero101111** [comment]
@g-glowacki Analyzing your sample files, I figured out what the problem is:
Your property sheets have item conditions written by hand. `clang-build.ps1` is currently looking for item conditions generated by Visual Studio (e.g. `'$config|$platform'=='Debug|Win32'`).

We're working on enabling `clang-build.ps1` to understand custom conditions, such as yours. 
Please keep in mind, however, that this is more than a trivial bug fix and will take some time to implement.

We'll keep you posted.
Thank you for your feedback.

**Comment by hero101111** [comment]
Hi @g-glowacki ,

We're happy to inform you that we've just released version 2.0 of [Clang Power Tools](https://marketplace.visualstudio.com/items?itemName=vs-publisher-690586.ClangPowerTools).
It fixes the project xml data [limitations](https://github.com/Caphyon/clang-power-tools/blob/master/CHANGELOG.md) that were at the core of this issue.

Your projects should now integrate successfully with Clang Power Tools.

The Visual Studio Extension should now update automatically to v2.0. 
If you've disabled automatic updates you'll have to update the extension manually.


Thank you for feedback and patience.

Regards,
Gabriel.

**Comment by g-glowacki** [comment]
I can confirm that it works fine now:) Thanks a lot!
